src/librustdoc/html/render.rs

```diff
@@ -325,11 +331,32 @@ impl Serialize for Type {
     }
 }
 
+/// A type used for the search index.
+#[derive(Debug)]
+struct Generic {
```

Comment:
It worries me that with this, we now have `Generic` and `Generics` and `generics: Generics` and `generics: Vec<Generic>`

Reply:
That seems correct to me? Do you know what's bugging you with this?

Reply:
Just that there's so many things which can be referred to as 'the generics' which have different types and possibly semantics.  If you're OK with this potential confusion then I won't block on this because I'm not really deep in this code like you are :D